mars/web/session.py

```diff
@@ -167,20 +168,36 @@ def fetch(self, *tileables, **kw):
 
         results = list()
         for tileable in tileables:
-            key = tileable.key
+            if tileable.key not in self._executed_tileables and isinstance(tileable.op, TensorIndex):
+                key = tileable.inputs[0].key
+                indexes = tileable.op.indexes
+                if not all(isinstance(ind, slice) for ind in indexes):
+                    raise ValueError('Only support fetch data slices')
+            else:
+                key = tileable.key
+                indexes = []
 
             if key not in self._executed_tileables:
                 raise ValueError('Cannot fetch the unexecuted tileable')
 
+            slice_str = '-'.join(['{0},{1},{2}'.format(s.start, s.stop, s.step) for s in indexes])
+
             session_url = self._endpoint + '/api/session/' + self._session_id
             compression_str = ','.join(v.value for v in dataserializer.get_supported_compressions())
-            data_url = session_url + '/graph/%s/data/%s?compressions=%s' \
-                % (self._get_tileable_graph_key(key), key, compression_str)
+            data_url = session_url + '/graph/%s/data/%s?compressions=%s&slices=%s' \
+                % (self._get_tileable_graph_key(key), key, compression_str, slice_str)
             resp = self._req_session.get(data_url, timeout=timeout)
             if resp.status_code >= 400:
                 raise ValueError('Failed to fetch data from server. Code: %d, Reason: %s, Content:\n%s' %
                                  (resp.status_code, resp.reason, resp.text))
-            results.append(dataserializer.loads(resp.content))
+            result_data = dataserializer.loads(resp.content)
+            if hasattr(tileable, 'index_value'):
```

Comment:
Seems that have seen the code below for a couple of times, maybe just making a util?